services/base.spec.js

```diff
@@ -42,6 +42,18 @@ class DummyService extends BaseService {
         staticExample: this.render({ namedParamA: 'foo', queryParamA: 'bar' }),
         keywords: ['hello'],
       },
+      {
+        pattern: ':world',
+        exampleUrl: 'World',
+        staticExample: this.render({ namedParamA: 'foo', queryParamA: 'bar' }),
+        keywords: ['hello'],
+      },
+      {
+        pattern: ':world',
+        namedParams: { world: 'World' },
+        staticExample: this.render({ namedParamA: 'foo', queryParamA: 'bar' }),
+        keywords: ['hello'],
+      },
```

Comment:
As the validation logic grows more complicated its all got a bit confusing it would be helpful to formalise the behaviour with some tests in here ensuring we're doing the right thing with various combinations of `previewUrl` / `exampleUrl` / `namedParams` etc. Moving the validation rules to their own function will also help with this.

Reply:
Do you mean tests of invalid examples? Yea, agreed.